tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,307 @@
+}
+
+func validateErrorsWhileCreatingCrtbBySteveWhenUserInputNotExistedRoleTemplateName(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["roleTemplateName"] = "roleTemplateNameNotExisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, "400 Bad Request")
+}
+
+func validateErrorsWhileCreatingCrtbBySteveWhenUserInputClusterNameAndNameSpaceNameDifferentAndNotExisted(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["clusterName"] = "local"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, "400 Bad Request")
+}
+
+func validateErrorsWhileCreatingCrtbBySteveWhenUserInputNotExistedNameSpaceName(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["metadata"].(map[string]interface{})["namespace"] = "namespacenotexisted"
+	inputData["clusterName"] = "namespacenotexisted"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, "404 Not Found")
+}
+
+func validateErrorsWhileCreatingCrtbBySteveWhenUserInputClusterNameAndNameSpaceNameDifferentAndExisted(t *testing.T, client *rancher.Client, remoteClusterID string, readJson []byte) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	require.NoError(t, err)
+	crtbName := namegen.RandStringLower(8)
+
+	inputData := unmarshalReadJson(t, readJson, remoteClusterID, crtbName, newUser.ID, defaultRoleTemplateName)
+	require.NoError(t, err)
+	inputData["clusterName"] = "local"
+
+	_, err = client.Steve.SteveType(crtbApiEndPoint).Create(inputData)
+	require.Error(t, err)
+	require.ErrorContains(t, err, "Forbidden: clusterName and namespace must be the same value")
+}
+
+func (crtb *CRTBGenTestSuite) TestCreateCRTB() {
+	tests := []struct {
+		roleTemplateName string
+	}{
+		{"cluster-owner"},
+		{"cluster-member"},
+		{"clusterroletemplatebindings-manage"},
+	}
+	for _, tt := range tests {
```

Comment:
If the test is written this way, does it mean that each test depends on the order? i.e. `validateCrtbCreationBySteveWithUserPrincipalNameAsInputAndItIsUsed` depends on `validateCrtbCreationBySteveWithAllRequiredInputValues` running first? If not, I thought we agreed with team 1 that tests would not be written this way unless explicitly requiring sequential tests.  cc @anupama2501 for clarification, maybe I am misremembering the conversation. But as I'm reading this, it would seem to make a lot more sense to have each of the function calls be tests in the suite instead, as they seem pretty independent of eachother.